src/symbol_object.cpp

```diff
@@ -34,6 +34,8 @@ StringObject *SymbolObject::inspect(Env *env) {
         if (c < 33 || c > 126) // FIXME: probably can be a smaller range
             quote = true;
     }
+    if (m_name.length() > 1 && m_name[0] == '$')
```

Comment:
Special case where symbols that start with `$` but are not exactly `:$` should get quotes.  This cleans up many failing specs for `#inspect`